oauth2_http/javatests/com/google/auth/oauth2/MockTokenServerTransport.java

```diff
@@ -206,7 +210,8 @@ public LowLevelHttpResponse execute() throws IOException {
             }
             String foundSecret = query.get("client_secret");
             String expectedSecret = clients.get(foundId);
-            if (foundSecret == null || !foundSecret.equals(expectedSecret)) {
+            if ((foundSecret == null || !foundSecret.equals(expectedSecret))
+                && clients.get("Authorization") == null) {
```

Comment:
I don't think we are validating the auth header anywhere.

Reply:
This is the limitation of `MockTokenServerTransport` as it doesn't capture request header during it's `execute` method. So we have to use an `addHeader` method to simulate what has been added to the headers. And then in the test code call we cal this `addHeader` prior to make the http request.

Reply:
You can get the request header -- we do it in other tests I'm pretty sure